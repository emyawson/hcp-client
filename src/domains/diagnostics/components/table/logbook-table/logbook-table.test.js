import { shallow } from 'enzyme';
import React from 'react';

import { LogbookTable } from '.';

const mockLogbookData = [
  {
    day: 'Fri, 17 Oct 2017',
    mealTimes: {
      NIGHT: [
        { date: '', bloodGlucose: 40, bolus: 35, carbohydrates: 15 },
        { date: '', bloodGlucose: 39, bolus: null, carbohydrates: null },
        { date: '', bloodGlucose: 40, bolus: null, carbohydrates: null },
      ],
      BEFORE_BREAKFAST: [
        { date: '', bloodGlucose: 40, bolus: 35, carbohydrates: 15 },
        { date: '', bloodGlucose: 40, bolus: 35, carbohydrates: 15 },
        { date: '', bloodGlucose: 40, bolus: 35, carbohydrates: 15 },
      ],
      AFTER_BREAKFAST: [
        { date: '', bloodGlucose: 40, bolus: 35, carbohydrates: 15 },
      ],
      BEFORE_LUNCH: [
        { date: '', bloodGlucose: 40, bolus: 35, carbohydrates: 15 },
      ],
      AFTER_LUNCH: [
        { date: '', bloodGlucose: 40, bolus: 35, carbohydrates: 15 },
      ],
      BEFORE_DINNER: [
        { date: '', bloodGlucose: 40, bolus: 35, carbohydrates: 15 },
      ],
      AFTER_DINNER: [
        { date: '', bloodGlucose: 40, bolus: 35, carbohydrates: 15 },
      ],
      BEDTIME: [{ date: '', bloodGlucose: 40, bolus: 35, carbohydrates: 15 }],
    },
  },
  {
    day: 'Sat, 18 Oct 2017',
    mealTimes: {
      NIGHT: [
        { date: '', bloodGlucose: 40, bolus: 35, carbohydrates: 15 },
        { date: '', bloodGlucose: 39, bolus: null, carbohydrates: null },
        { date: '', bloodGlucose: 40, bolus: null, carbohydrates: null },
      ],
      BEFORE_BREAKFAST: [
        { date: '', bloodGlucose: 40, bolus: 35, carbohydrates: 15 },
      ],
      AFTER_BREAKFAST: [
        { date: '', bloodGlucose: 40, bolus: 35, carbohydrates: 15 },
      ],
      BEFORE_LUNCH: [
        { date: '', bloodGlucose: 40, bolus: 35, carbohydrates: 15 },
      ],
      AFTER_LUNCH: [
        { date: '', bloodGlucose: 40, bolus: 35, carbohydrates: 15 },
      ],
      BEFORE_DINNER: [
        { date: '', bloodGlucose: 40, bolus: 35, carbohydrates: 15 },
      ],
      AFTER_DINNER: [
        { date: '', bloodGlucose: 40, bolus: 35, carbohydrates: 15 },
      ],
      BEDTIME: [{ date: '', bloodGlucose: 40, bolus: 35, carbohydrates: 15 }],
    },
  },
];

describe('LogbookTable test suite', () => {
  test('LogbookTable renders correctly', () => {
    const wrapper = shallow(<LogbookTable logbookData={mockLogbookData} />);
    expect(wrapper).toMatchSnapshot();
  });
});
