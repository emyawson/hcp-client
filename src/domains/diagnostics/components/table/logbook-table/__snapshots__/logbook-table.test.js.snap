// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogbookTable test suite LogbookTable renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    logbookData={
      Array [
        Object {
          "day": "Fri, 17 Oct 2017",
          "mealTimes": Object {
            "AFTER_BREAKFAST": Array [
              Object {
                "bloodGlucose": 40,
                "bolus": 35,
                "carbohydrates": 15,
                "date": "",
              },
            ],
            "AFTER_DINNER": Array [
              Object {
                "bloodGlucose": 40,
                "bolus": 35,
                "carbohydrates": 15,
                "date": "",
              },
            ],
            "AFTER_LUNCH": Array [
              Object {
                "bloodGlucose": 40,
                "bolus": 35,
                "carbohydrates": 15,
                "date": "",
              },
            ],
            "BEDTIME": Array [
              Object {
                "bloodGlucose": 40,
                "bolus": 35,
                "carbohydrates": 15,
                "date": "",
              },
            ],
            "BEFORE_BREAKFAST": Array [
              Object {
                "bloodGlucose": 40,
                "bolus": 35,
                "carbohydrates": 15,
                "date": "",
              },
              Object {
                "bloodGlucose": 40,
                "bolus": 35,
                "carbohydrates": 15,
                "date": "",
              },
              Object {
                "bloodGlucose": 40,
                "bolus": 35,
                "carbohydrates": 15,
                "date": "",
              },
            ],
            "BEFORE_DINNER": Array [
              Object {
                "bloodGlucose": 40,
                "bolus": 35,
                "carbohydrates": 15,
                "date": "",
              },
            ],
            "BEFORE_LUNCH": Array [
              Object {
                "bloodGlucose": 40,
                "bolus": 35,
                "carbohydrates": 15,
                "date": "",
              },
            ],
            "NIGHT": Array [
              Object {
                "bloodGlucose": 40,
                "bolus": 35,
                "carbohydrates": 15,
                "date": "",
              },
              Object {
                "bloodGlucose": 39,
                "bolus": null,
                "carbohydrates": null,
                "date": "",
              },
              Object {
                "bloodGlucose": 40,
                "bolus": null,
                "carbohydrates": null,
                "date": "",
              },
            ],
          },
        },
        Object {
          "day": "Sat, 18 Oct 2017",
          "mealTimes": Object {
            "AFTER_BREAKFAST": Array [
              Object {
                "bloodGlucose": 40,
                "bolus": 35,
                "carbohydrates": 15,
                "date": "",
              },
            ],
            "AFTER_DINNER": Array [
              Object {
                "bloodGlucose": 40,
                "bolus": 35,
                "carbohydrates": 15,
                "date": "",
              },
            ],
            "AFTER_LUNCH": Array [
              Object {
                "bloodGlucose": 40,
                "bolus": 35,
                "carbohydrates": 15,
                "date": "",
              },
            ],
            "BEDTIME": Array [
              Object {
                "bloodGlucose": 40,
                "bolus": 35,
                "carbohydrates": 15,
                "date": "",
              },
            ],
            "BEFORE_BREAKFAST": Array [
              Object {
                "bloodGlucose": 40,
                "bolus": 35,
                "carbohydrates": 15,
                "date": "",
              },
            ],
            "BEFORE_DINNER": Array [
              Object {
                "bloodGlucose": 40,
                "bolus": 35,
                "carbohydrates": 15,
                "date": "",
              },
            ],
            "BEFORE_LUNCH": Array [
              Object {
                "bloodGlucose": 40,
                "bolus": 35,
                "carbohydrates": 15,
                "date": "",
              },
            ],
            "NIGHT": Array [
              Object {
                "bloodGlucose": 40,
                "bolus": 35,
                "carbohydrates": 15,
                "date": "",
              },
              Object {
                "bloodGlucose": 39,
                "bolus": null,
                "carbohydrates": null,
                "date": "",
              },
              Object {
                "bloodGlucose": 40,
                "bolus": null,
                "carbohydrates": null,
                "date": "",
              },
            ],
          },
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "children": [Function],
      "disableHeight": false,
      "disableWidth": false,
      "onResize": [Function],
      "style": Object {},
    },
    "ref": null,
    "rendered": [Function],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": [Function],
        "disableHeight": false,
        "disableWidth": false,
        "onResize": [Function],
        "style": Object {},
      },
      "ref": null,
      "rendered": [Function],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
